ewatercycle/models/abstract.py

```diff
@@ -150,3 +168,20 @@ def time_step(self) -> float:
     def output_var_names(self) -> Iterable[str]:
         """List of a model's output variables."""
         return self.bmi.get_output_var_names()
+
+    def _check_parameter_set(self):
+        if not self.parameter_set:
+            # Nothing to check
+            return
+        model_name = self.__class__.__name__.lower()
+        if model_name != self.parameter_set.target_model:
+            raise ValueError(f'Parameter set has wrong target model, '
+                             f'expected {model_name} got {self.parameter_set.target_model}')
+        if self.parameter_set.supported_model_versions == set():
+            logger.warning(f'Model expects parameter set to support version {self.version}, '
+                           f'but parameter set supports any version')
```

Comment:
If the parameter set supports ("should work with"?) any version, that should be just fine, right? Why issue a warning?

Reply:
Can we use "compatible with" instead of "x supports y"?

Reply:
I wanted to give a warning to the user that parameter set is potentially untested with the selected model version. Maybe better to lower it to `logger.info`

Reply:
```suggestion
        if self.parameter_set.supported_model_versions == set():
            logger.info(f'Model version {self.version} is not explicitly listed in the supported model versions of this parameter set. This can lead to compatibility issues.')
```